src/hooks/useArweaveService.ts

```diff
@@ -11,13 +13,37 @@ const initArweave = () => {
   });
 };
 
-const useArchaeologistService = () => {
+const useArweaveService = () => {
   const [sendStatus, setStatus] = useState<{ status: string; confirmations: number }>({
     status: '',
     confirmations: 0,
   });
+
+  const downloadArweaveFile = async (
```

Comment:
Can we name this `isArweaveFileValid`? Since it returns a boolean, would help clarify the purpose of the function.

The logic in here could be refactored into a `fetchAndDecryptArweaveFile` function which returns the decrypted file bytes.

Then this function could validate those file bytes match the original file bytes.

We can then also use the `fetchAndDecryptArweaveFile` function for the `resurrectFile` function which should download the decrypted file bytes to the user's computer. This file should be downloaded with the original file extension it was uploaded with (`.txt`, etc)

Reply:
yeah that function can be done later for sure -- I will create an issue for it